slsim/lens_pop.py

```diff
@@ -1,378 +1,70 @@
-                )
-            else:
-                from slsim.Sources.SupernovaeCatalog.supernovae_sample import (
-                    SupernovaeCatalog,
-                )
-
-                suffixes = []
-                for key in kwargs_variability:
-                    if key.startswith("ps_mag_"):
-                        suffixes.append(key.split("ps_mag_")[1])
-                    elif len(key) == 1:
-                        suffixes.append(key)
-                supernovae_catalog_class = SupernovaeCatalog(
-                    sn_type=sn_type,
-                    band_list=suffixes,
-                    lightcurve_time=lightcurve_time,
-                    absolute_mag=None,
-                    absolute_mag_band=sn_absolute_mag_band,
-                    mag_zpsys=sn_absolute_zpsys,
-                    cosmo=cosmo,
-                    skypy_config=skypy_config,
-                    sky_area=self.source_sky_area,
-                    sn_modeldir=sn_modeldir,
-                )
-                if source_type == "supernovae":
-                    supernovae_sample = supernovae_catalog_class.supernovae_catalog(
-                        host_galaxy=False, lightcurve=False
-                    )
-                    self._sources = PointSources(
-                        supernovae_sample,
-                        cosmo=cosmo,
-                        sky_area=self.source_sky_area,
-                        kwargs_cut=kwargs_source_cut,
-                        variability_model=variability_model,
-                        kwargs_variability_model=kwargs_variability,
-                        light_profile=source_light_profile,
-                    )
-                else:
-                    supernovae_sample = supernovae_catalog_class.supernovae_catalog(
-                        host_galaxy=True, lightcurve=False
-                    )
-                    self._sources = PointPlusExtendedSources(
-                        supernovae_sample,
-                        cosmo=cosmo,
-                        sky_area=self.source_sky_area,
-                        kwargs_cut=kwargs_source_cut,
-                        variability_model=variability_model,
-                        kwargs_variability_model=kwargs_variability,
-                        list_type="astropy_table",
-                        light_profile=source_light_profile,
-                    )
-            if source_type == "supernovae":
-                self._source_model_type = "point_source"
-            else:
-                self._source_model_type = "point_plus_extended"
-        else:
-            raise ValueError("source_type %s is not supported" % source_type)
+        self._lens_galaxies = deflector_population
+        self._sources = source_population
 
         self._factor_source = self.f_sky.to_value(
             "deg2"
-        ) / self.source_sky_area.to_value("deg2")
+        ) / self._sources._sky_area.to_value("deg2")
```

Comment:
we should avoid accessing private variables of other classes. Use self._sources.sky_area instead